src/main/java/pl/pjwstk/woloappapi/config/SecurityConfig.java

```diff
@@ -0,0 +1,73 @@
+package pl.pjwstk.woloappapi.config;
+
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.security.SecurityProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.env.Environment;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
+import org.springframework.security.oauth2.core.ClientAuthenticationMethod;
+import org.springframework.security.oauth2.core.oidc.IdTokenClaimNames;
+import org.springframework.security.web.SecurityFilterChain;
+import pl.pjwstk.woloappapi.repository.UserRepository;
+import pl.pjwstk.woloappapi.service.CustomUserDetailsService;
+
+import static org.springframework.security.config.Customizer.withDefaults;
+@AllArgsConstructor
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig {
+    private final UserRepository userRepository;
+
+    @Autowired
+    public PasswordEncoder passwordEncoder;
+
+    @Bean
+    @Order(SecurityProperties.BASIC_AUTH_ORDER)
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        http
+                .authorizeHttpRequests((requests) -> requests
+                        .requestMatchers("/secured").hasAuthority("Użytkownik")
+                        .requestMatchers(HttpMethod.POST, "/register").permitAll()
+                        .requestMatchers(HttpMethod.POST, "/login").permitAll()
+                        .anyRequest().permitAll()
+                )
+                .oauth2Login(withDefaults())
+                .formLogin(withDefaults());
+
+        return http.build();
+    }
+    @Bean
+    public UserDetailsService userDetailsService(UserRepository userRepository) {
+        return new CustomUserDetailsService(userRepository);
+    }
+
+
+    public ClientRegistration googleClientRegistration() {
+
+
+        return ClientRegistration.withRegistrationId("google")
+                .clientId("13893580502-b542hnte437qe2froi35flpd8dd83u1o.apps.googleusercontent.com")
+                .clientSecret("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")
```

Comment:
Secrets shouldn't be visible in code

Reply:
funkcja nie była wykorzystywana więc została usunięta, application.properties dodane do git ignore. W przyszłości client secret będzie wczytywany poprzez inject do env.